src/sortStudents.ts

```diff
@@ -1,16 +1,98 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+const averageStudentGrade = (student: Student):number => {
+  return student
+    .grades.reduce((accum: number, grade: number): number => {
+      return accum + grade;
+    }, 0) / student.grades.length;
+};
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const resArr: Student[] = [...students];
+  const sortOrder: boolean = order === 'asc';
+
+  switch (sortBy) {
+    case SortType.Name:
+    case SortType.Surname:
+      resArr.sort((firstStudent: Student, secondStudent: Student): number => {
+        return sortOrder
+          ? firstStudent[sortBy].localeCompare(secondStudent[sortBy])
+          : secondStudent[sortBy].localeCompare(firstStudent[sortBy]);
+      });
+      break;
+    case SortType.Age:
+      resArr.sort((firstStudent: Student, secondStudent: Student): number => {
+        let res = 0;
+
+        if (firstStudent.age > secondStudent.age) {
+          res = sortOrder ? 1 : -1;
+        }
+
+        if (firstStudent.age < secondStudent.age) {
+          res = !sortOrder ? 1 : -1;
+        }
+
+        return res;
```

Comment:
You don't need `res` variable, you could simply write this way 👇 
```suggestion
        return sortOrder
          ? firstStudent.age - secondStudent.age
          : secondStudent.age - firstStudent.age;
```

for `SortType.Married and SortType.AverageGrade` the same but addapt it ☝️